### DISCUSSION TITLE
Will Timefold support multiple PlanningListVariables in the future?

### DISCUSSION THREAD

**Discussion by kentbill** [TimefoldAI/timefold-solver#1141] [General]
Hell, Teams,

I'm doing a project similar to FieldServiceRouting, but I need to schedule not only the technician but a team with technicians. So, Besides the job scheduling, I also want to assign the technician to a team. The technician has different qualifications or skills. A team can provide services determined by the technicians in the team. 
For the modeling design, I define the team as PlanningEntity, which contains two PlanningListVariables - **jobList** and **technicianList**. Only the jobList needs to be routed, and the technicianList in the team is just a list, the order of the technicianList in a team is not important.
But I get an UnsupportedOperationException:
**Exception in thread "main" java.lang.UnsupportedOperationException: Defining multiple list variables ([Resource.jobList variable, Resource.personList variable]) across the model is currently not supported.**
﻿
In future versions of Timefold, are you considering supporting multiple PlanningListVariables?

**Comment by triceo** [comment]
Hello @kentbill! When it comes to new features, I never say never, and we are aware that this is something that users sometimes ask for. But considering what we have on our plate right now, I think it's fair to say that we won't be working on this any time soon.

As a side note, it's not very clear what "multiple list variables" would even mean. If your planning variable doesn't require ordering, then you don't need a list variable at all - list variable only exists to model a precedence relationship, X comes before Y.

**Reply by kentbill** [reply]
Got it, thank you, @triceo 
I think the annotation @InverseRelationShadowVariable could help me in this case.
